src/app/components/SubjectHeading/Search/SubjectHeadingSearch.jsx

```diff
@@ -33,14 +33,27 @@ class SubjectHeadingSearch extends React.Component {
 
     this.setState({
       userInput,
-    }, () => {
-      axios(`${appConfig.baseUrl}/api/subjectHeadings/autosuggest?query=${userInput}`)
+    }, this.makeApiCallWithThrottle(userInput));
+  }
+
+  makeApiCallWithThrottle(userInput) {
+    const apiCall = () => {
+      return axios(`${appConfig.baseUrl}/api/subjectHeadings/autosuggest?query=${userInput}`)
         .then((res) => {
-          this.setState({
+          if (res.data.request.query === this.state.userInput) this.setState({
             suggestions: res.data.autosuggest,
-          });
+            activeSuggestion: 0,
+          })
         });
-    });
+    };
+
+    let timerId;
+
+    if (timerId || userInput.length === 0) return;
+    timerId = setTimeout(() => {
+      apiCall();
+      timerId = undefined;
+    }, 500);
```

Comment:
@danamansana here it is given a value. Also, added a new commit.

Reply:
Yes but I wouldn't think that value of timerId would persist to the next function invocation

Reply:
Ahh. You are correct.